### PR TITLE
MEN-5241: Follow ups

### PR DESCRIPTION
**Pull request by oleorhagen**
* Add the template to the static check as well
* Run the check on all branches, not dicounting released, hosted and staging branches.

### CODE REVIEW

**Review by lluiscampos — APPROVED** [review]
Thanks!

For release branches of `mender` and `mender-artifact`, we need to add `.gitlab-ci-check-golang-static.yml` to their pipelines [here](https://github.com/mendersoftware/mender/blob/3.1.x/.gitlab-ci.yml#L7) and [here](https://github.com/mendersoftware/mender-artifact/blob/3.6.x/.gitlab-ci.yml#L7).

File: .gitlab-ci-check-golang-lint.yml [hunk 1]
```diff
@@ -46,8 +46,6 @@ test:govendor-check:
   needs: []
   variables:
     GOLANG_VERSION: "1.16"
-  except:
-    - /^(master|hosted|staging|production|[0-9]+\.[0-9]+\.x|^saas-[a-zA-Z0-9.]+)$/
```
Comment (mzedel) [inline .gitlab-ci-check-golang-lint.yml:50]: while possible for release branches, IMO no change should land in `hosted|staging|production|^saas-[a-zA-Z0-9.]+` that didn't go through the master pipeline first - so we could skip these runner minutes there

**Review by mzedel — APPROVED** [review]
Thank you!

File: .gitlab-ci-check-golang-lint.yml (hunk 1 above)
Comment (oleorhagen) [reply .gitlab-ci-check-golang-lint.yml:50]: Better safe than sorry? I mean, the pipe is literally 10 seconds

Comment (mzedel) [reply .gitlab-ci-check-golang-lint.yml:50]: 🚀 in that case, sure - run it everywhere 🎉

**Pull request merged**